### PR TITLE
Feat/add firebase databse

### PR DESCRIPTION
**Pull request by jhwan7**

### CODE REVIEW

**Comment by jhwan7** [comment]
i added a sample screen where i showed examples of how some of the tests work. Also, i made functions that i could only think of, if we end up needing more functions then ill add them later

File: app/utilities/firebase/firebase.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import firebase from "firebase"
+import firebaseConfig from "../../env/firebaseConfig.json"
+
+  // Initialize Firebase
+  firebase.initializeApp(firebaseConfig);
```
Comment (jsong336) [inline app/utilities/firebase/firebase.tsx:5]: `firebase.apps.length ? firebase.app() : firebase.initializeApp(firebaseConfig)`

Comment (jhwan7) [reply app/utilities/firebase/firebase.tsx:5]: what does this do?

Comment (jhwan7) [reply app/utilities/firebase/firebase.tsx:5]: just a safety precaution?

File: app/utilities/firebase/firebaseActions.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    [ROOM_KEY.QNA]: "",
+    [ROOM_KEY.CURR_ROOM_SIZE]: 0
+  }).then (snap => {
+    return snap.key;
+  });
+}
+
+export function getCurrentRoomSize (roomKey: string) {
+  var roomRef = firebase.database().ref(`${roomKey}`)
+  return roomRef.get()
```
Comment (jsong336) [inline app/utilities/firebase/firebaseActions.tsx:20]: just curiosity, is `get()` different from `once('value', (snapshot) => ...` ?

File: app/utilities/firebase/firebase.tsx (hunk 1 above)
Comment (jsong336) [reply app/utilities/firebase/firebase.tsx:5]: yeah I think I've seen this a lot more. I think it just make sure we don't create app twice

**Review by jsong336 — APPROVED** [review]
Seems good 👍

File: app/utilities/firebase/firebaseActions.tsx (hunk 2 above)
Comment (jhwan7) [reply app/utilities/firebase/firebaseActions.tsx:20]: i think once() gets executed when the db changes, (child most likely). Whereas get is just called instantly when the method is called

**Comment by jhwan7** [comment]
> Seems good 👍
> @jhwan7 Does squash merge remove the commit with credential history?

maybe makes it harder to find

**Pull request merged**